usersession/agent/rest_api.go

```diff
@@ -433,8 +460,8 @@ func postPendingRefreshNotification(c *Command, r *http.Request) Response {
 	return SyncResponse(nil)
 }
 
-func guessAppIcon(si *snap.Info) string {
-	var icon string
+func guessAppData(si *snap.Info, defaultName string) (string, string) {
+	var icon, name string
```

Comment:
Consider simplifying this to:

```suggestion
func guessAppData(si *snap.Info, defaultName string) (icon string, name string) {
```

Keep the `return icon, name` below though, since there is a static checker in place to forbid naked returns.

Reply:
I did that change, but now the static checker is failing. And yes, I kept the "return icon, name" part (although also tried without, just in case...). The error is `Error: If the the action is run outside a pull request it should be run with a filelist.`, but google says nothing about it...

Reply:
I was referring to the nakedret checks [from the run-checks script](https://github.com/snapcore/snapd/blob/ce451ceba0cac95c8727bf33f91ce788eee315cc/run-checks#L289-L301) in the "Tests / static-checks" jobs.

The failure you're seeing is in a new workflow that seems was added 2 days ago, and appears to be failing for everyone:

https://github.com/snapcore/snapd/actions/workflows/static-analysis.yaml

From the look of it, it relies on a secret that won't be available to pull request CI runs. I don't think there is anything you can do to address this failure in your branch.

Reply:
Oh, so this is what you were talking this morning! I hadn't read this until now :-)